uniffi_bindgen/src/templates/ObjectTemplate.rs

```diff
@@ -1,36 +1,36 @@
 // For each Object definition, we assume the caller has provided an appropriately-shaped `struct`
-// with an `impl` for each method on the object. We create a `ConcurrentHandleMap` for safely handing
+// with an `impl` for each method on the object. We create an `Arc` in
+// `[Threadsafe]` interfaces or `Arc<Mutex<>>` otherwise, for for "safely" handing
 // out references to these structs to foreign language code, and we provide a `pub extern "C"` function
 // corresponding to each method.
 //
+// Note that "safely" is in "scare quotes" - that's because we use functions on
+// an `Arc` that are inherently unsafe, but the code we generate is safe in
+// practice.
+//
 // If the caller's implementation of the struct does not match with the methods or types specified
 // in the UDL, then the rust compiler will complain with a (hopefully at least somewhat helpful!)
 // error message when processing this generated code.
-{% let handle_map = format!("UNIFFI_HANDLE_MAP_{}", obj.name().to_uppercase()) -%}
-
-uniffi::deps::lazy_static::lazy_static! {
-    {%- let handle_map_type = obj.threadsafe()|choose(
-        "uniffi::ffi::handle_maps::ArcHandleMap",
-        "uniffi::ffi::handle_maps::MutexHandleMap")
-    %}
-    #[doc(hidden)]
-    static ref {{ handle_map }}: {{ handle_map_type }}<{{ obj.name() }}>
-        = Default::default();
-}
 
-    {% let ffi_free = obj.ffi_object_free() -%}
-    #[doc(hidden)]
-    #[no_mangle]
-    pub extern "C" fn {{ ffi_free.name() }}(handle: u64) {
-        let _ = {{ handle_map }}.delete_u64(handle);
+{% let ffi_free = obj.ffi_object_free() -%}
+#[doc(hidden)]
+#[no_mangle]
+pub extern "C" fn {{ ffi_free.name() }}(ptr: u64) {
```

Comment:
Yeah, I really think we want to update the `FFIType` enum to let us use a pointer type in the argument here.